### ISSUE TITLE
Netty HTTP/2 not working

### ISSUE THREAD

**Issue by icode** [ktorio/ktor#1261]
The problem reproduces with http2-push from https://github.com/ktorio/ktor-samples

Run ./gradlew :http2-push:run

```kotlin
const val kotlin = "1.3.41"
const val ktor = "1.2.3"
```

give a exception

```
Exception in thread "main" java.lang.NoClassDefFoundError: io.netty.handler.ssl.OpenSsl (initialization failure)
	at java.base/java.lang.J9VMInternals.initializationAlreadyFailed(J9VMInternals.java:143)
	at io.netty.handler.ssl.SslContext.defaultProvider(SslContext.java:120)
	at io.netty.handler.ssl.SslContext.defaultServerProvider(SslContext.java:107)
	at io.netty.handler.ssl.SslContext.newServerContextInternal(SslContext.java:454)
	at io.netty.handler.ssl.SslContextBuilder.build(SslContextBuilder.java:457)
	at io.ktor.server.netty.NettyChannelInitializer.<init>(NettyChannelInitializer.kt:72)
	at io.ktor.server.netty.NettyApplicationEngine.<init>(NettyApplicationEngine.kt:103)
	at io.ktor.server.netty.Netty.create(Embedded.kt:14)
	at io.ktor.server.netty.Netty.create(Embedded.kt:12)
	at io.ktor.server.engine.EmbeddedServerKt.embeddedServer(EmbeddedServer.kt:79)
	at io.ktor.server.engine.EmbeddedServerKt.embeddedServer$default(EmbeddedServer.kt:77)
	at io.ktor.samples.http2push.MainKt.main(Main.kt:17)
Caused by: java.lang.NoSuchMethodError: io/netty/internal/tcnative/SSLContext.setCipherSuite(JLjava/lang/String;Z)Z (loaded from file:/Users/icode/.gradle/caches/modules-2/files-2.1/io.netty/netty-tcnative/2.0.7.Final/b9413ea71da2d14e6f5644aa33a5c4d966090540/netty-tcnative-2.0.7.Final.jar by jdk.internal.loader.ClassLoaders$AppClassLoader@760c26b8) called from class io.netty.handler.ssl.OpenSsl (loaded from file:/Users/icode/.gradle/caches/modules-2/files-2.1/io.netty/netty-handler/4.1.37.Final/a242f98517b4dc1a53a0eb7dad8ad30d4f4b9f2e/netty-handler-4.1.37.Final.jar by jdk.internal.loader.ClassLoaders$AppClassLoader@760c26b8).
	at io.netty.handler.ssl.OpenSsl.<clinit>(OpenSsl.java:201)
	at io.ktor.server.netty.NettyChannelInitializer$Companion.findAlpnProvider(NettyChannelInitializer.kt:158)
	at io.ktor.server.netty.NettyChannelInitializer$Companion.access$findAlpnProvider(NettyChannelInitializer.kt:147)
	at io.ktor.server.netty.NettyChannelInitializer$Companion$alpnProvider$2.invoke(NettyChannelInitializer.kt:148)
	at io.ktor.server.netty.NettyChannelInitializer$Companion$alpnProvider$2.invoke(NettyChannelInitializer.kt:147)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at io.ktor.server.netty.NettyChannelInitializer$Companion.getAlpnProvider$ktor_server_netty(NettyChannelInitializer.kt)
	at io.ktor.server.netty.NettyChannelInitializer.<init>(NettyChannelInitializer.kt:58)
	... 6 more

```

I update `tcnative_version` to `2.0.25.Final` , then start is ok
I get this exception when request
Navigate to https://localhost:8443/
```
2019-08-02 20:23:13.309 [nioEventLoopGroup-3-2] WARN  i.n.h.s.ApplicationProtocolNegotiationHandler - [id: 0x00f04f96, L:/0:0:0:0:0:0:0:1:443 - R:/0:0:0:0:0:0:0:1:53289] TLS handshake failed:
javax.net.ssl.SSLHandshakeException: error:10000416:SSL routines:OPENSSL_internal:SSLV3_ALERT_CERTIFICATE_UNKNOWN
	at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.shutdownWithError(ReferenceCountedOpenSslEngine.java:965)
	at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.sslReadErrorResult(ReferenceCountedOpenSslEngine.java:1231)
	at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.unwrap(ReferenceCountedOpenSslEngine.java:1185)
	at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.unwrap(ReferenceCountedOpenSslEngine.java:1256)
	at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.unwrap(ReferenceCountedOpenSslEngine.java:1299)
	at io.netty.handler.ssl.SslHandler$SslEngineType$1.unwrap(SslHandler.java:204)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1329)
	at io.netty.handler.ssl.SslHandler.decodeNonJdkCompatible(SslHandler.java:1236)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1273)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:500)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:439)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1408)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:930)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:697)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:632)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:549)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:511)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
2019-08-02 20:23:13.310 [nioEventLoopGroup-3-2] WARN  i.n.channel.DefaultChannelPipeline - An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.handler.codec.DecoderException: javax.net.ssl.SSLHandshakeException: error:10000416:SSL routines:OPENSSL_internal:SSLV3_ALERT_CERTIFICATE_UNKNOWN
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:470)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1408)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:930)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:697)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:632)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:549)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:511)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
Caused by: javax.net.ssl.SSLHandshakeException: error:10000416:SSL routines:OPENSSL_internal:SSLV3_ALERT_CERTIFICATE_UNKNOWN
	at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.shutdownWithError(ReferenceCountedOpenSslEngine.java:965)
	at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.sslReadErrorResult(ReferenceCountedOpenSslEngine.java:1231)
	at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.unwrap(ReferenceCountedOpenSslEngine.java:1185)
	at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.unwrap(ReferenceCountedOpenSslEngine.java:1256)
	at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.unwrap(ReferenceCountedOpenSslEngine.java:1299)
	at io.netty.handler.ssl.SslHandler$SslEngineType$1.unwrap(SslHandler.java:204)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1329)
	at io.netty.handler.ssl.SslHandler.decodeNonJdkCompatible(SslHandler.java:1236)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1273)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:500)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:439)
	... 17 common frames omitted
```

**Comment by cy6erGn0m** [comment]
Address already in use

So you have wrong config (same port for different connectors) or there is another application that holds the port. 

Push is entirely irrelevant

**Comment by icode** [comment]
> 地址已在使用中
> 
> 所以你有错误的配置（不同连接器的相同端口）或另一个持有端口的应用程序。
> 
> 推是完全无关紧要的

I sent the wrong message and it has been changed.

**Comment by coder82** [comment]
t=275253 [st=3]        HTTP2_STREAM_ERROR
                       --> description = "Server reset stream."
                       --> net_error = "ERR_SPDY_PROTOCOL_ERROR"
                       --> stream_id = 71
t=275253 [st=3]     -HTTP_TRANSACTION_READ_HEADERS
                     --> net_error = -337 (ERR_SPDY_PROTOCOL_ERROR)
t=275253 [st=3]   -URL_REQUEST_START_JOB
                   --> net_error = -337 (ERR_SPDY_PROTOCOL_ERROR)

**Comment by yay** [comment]
I get `ERR_HTTP2_PROTOCOL_ERROR` on all server requests but the first one.
Please see a minimal project that reproduces this:
https://github.com/yay/ktor-netty-http2-issue

After running the app the first time, make sure to add the auto-generated certificate to Trusted Root Certification Authorities and re-run.

First load:
![first-load](https://user-images.githubusercontent.com/1799756/65375864-af3eae00-dc91-11e9-972a-7296d365979a.png)

Subsequent requests:
![post-reload](https://user-images.githubusercontent.com/1799756/65375869-b796e900-dc91-11e9-89a9-5a9a849a29da.png)

Windows 10 1903
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.4+11)
Kotlin 1.3.50
ktor 1.2.4
tcnative windows-x86_64 2.0.25.Final

**Comment by coder82** [comment]
Yes it needs to be fixed.

**Comment by corecanarias** [comment]
Same issue here, it works fine until you configure ssl certificates, then it seems like http2 gets automatically enabled and GET requests stop working, POST requests seems to be working though

**Comment by Benjozork** [comment]
Same issue here.
Is there any way to force HTTP1.1 in the meantime ?

**Comment by oleg-larshin** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/KTOR-578">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.

**Comment by e5l** [comment]
Fixed in `2.0.1`
